File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode curr = head;
        if (head == null)
        {
            return head;
        }
        for (int i=0;i<n;i++)
        {
            if (curr == null)
            {
                return curr;
            }
            curr = curr.next;
        }
        if (curr == null)
        {
            return head.next;
        }
        ListNode first = head;
        while (curr.next != null)
        {
            first = first.next;
            curr = curr.next;
        }
        first.next = first.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head == None:
            return head
        elif head.next == None and n == 1:
            return None
        l = 0
        curr = head
        curr2 = head
        while curr != None:
            l = l+1
            curr = curr.next
        if l == n:
            return head.next
        i = l - n
        while i > 1:
            head = head.next
            i = i - 1
        head.next = head.next.next
        return curr2